pymeasure/instruments/teledyne/teledyneHDO6xxx.py

```diff
@@ -0,0 +1,45 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2023 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.teledyne.teledyneMAUI import TeledyneMAUI, TeledyneMAUIChannel
+
+
+class TeledyneHDO6xxxChannel(TeledyneMAUIChannel):
+    """Extended Channel object for :class:`TeledyneHDO6xxx`."""
+    pass
+
+
+class TeledyneHDO6xxx(TeledyneMAUI):
+    """Reference to the Teledyne-Lecroy HDo6xxx class of oscilloscopes.
+
+    All functionality is inherited from the base class.
+    """
+
+    ch_1 = Instrument.ChannelCreator(TeledyneHDO6xxxChannel, 1)
+
+    ch_2 = Instrument.ChannelCreator(TeledyneHDO6xxxChannel, 2)
+
+    ch_3 = Instrument.ChannelCreator(TeledyneHDO6xxxChannel, 3)
+
+    ch_4 = Instrument.ChannelCreator(TeledyneHDO6xxxChannel, 4)
```

Comment:
Isn't that superfluous or will a later implementation differ from the MAUI version?

I see two options: state somewhere a list of oscilloscopes, which can be used by the MAUI class. The list could start with this one.

Another option is the following:
```suggestion
class TeledyneHDO6xxx(TeledyneMAUI):
    """Reference to the Teledyne-Lecroy HDo6xxx class of oscilloscopes.

    All functionality is inherited from the base class.
    """

    pass
```

Reply:
Yeah, you're right. With this approach (being consistent) we would have to add a new empty class for each supported device. Best to keep a list then.